src/utils/check-session.ts

```diff
@@ -52,11 +56,46 @@ export const checkUserSession = async ({
 		});
 
 		if (userData?.statusCode === apiStatusCodes.API_STATUS_UNAUTHORIZED) {
+			const refreshSession = getFromCookies(cookies, 'refresh');
+
+			const configRefreshToken = {
+				headers: {
+					'Content-Type': 'application/json'
+				},
+				method: 'POST',
+				body: JSON.stringify({
+					refreshToken: refreshSession
+				})
+			};
+
+			const res = await fetch(`${baseURL + apiRoutes.auth.refreshToken}`, {
+				...configRefreshToken,
+			});
+			const userSession = await res.json();
+	
+			if (userSession?.statusCode !== apiStatusCodes.API_STATUS_SUCCESS) {
+				return {
+					permitted: false,
+					redirect: pathRoutes.auth.sinIn,
+					authorized: false,
+				};
+			}
+			setToCookies(cookies, "session", userSession?.data?.access_token as string, {
+				path: "/"
+			})
+			setToCookies(cookies, "refresh", userSession?.data?.refresh_token as string, {
+				path: "/"
+			})
+
 			return {
-				permitted: false,
-				redirect: pathRoutes.auth.sinIn,
-				authorized: false,
+				permitted: true,
+				authorized: true,
+				sessionData: {
```

Comment:
If we are not using the `sessionData` object then can we remove this?

Reply:
Resolved